report/verification/mes/tmap_1e.md

```diff
@@ -0,0 +1,302 @@
+
+$$
+C = C_0 \left\{ \frac{(a-x) D_{SiC} + l D_{PyC}}{l D_{PyC} + a D_{SiC}} + 2 \sum_{n=1}^{\infty} B_n \sin\left(\lambda_n \frac{x}{a}\right) \exp\left(-D_{PyC} \frac{\lambda^2_n}{a^2} t \right) \right\}
+$$
+
+and
+
+$$
+C = C_0 \left\{ \frac{(l+a-x) D_{PyC}}{l D_{PyC} + a D_{SiC}} + 2 \sum_{n=1}^{\infty} B_n \frac{\sin(\lambda_n)}{\sin(k \lambda_n l/a)} \sin\left(k \lambda_n \frac{l+a-x}{a}\right) \exp\left(-D_{PyC} \frac{\lambda^2_n}{a^2} t \right) \right\}
+$$
+
+where
+
+$$
+k = \sqrt{\frac{D_{PyC}}{D_{SiC}}},
+$$
+
+$$
+B_n = \frac{D_{PyC} l \sin^2(k \lambda_n l/a) (\cos(\lambda_n) - 1) + D_{SiC} \sin(k \lambda_n l/a) (k l \sin(\lambda_n) \cos(k \lambda_n l/a) - a \sin(k \lambda_n l/a))}{ \lambda_n (a D_{SiC} + l D_{PyC}) (\sin^2(k \lambda_n l/a) + (l/a) \sin^2(\lambda_n))},
+$$
+
+and $\lambda_n$ are the roots of
+
+$$
+\frac{\sin(\lambda_n) \cos(k \lambda_n l/a)}{k} + \cos(\lambda_n) \sin(k \lambda_n l/a) = 0.
+$$
+
+```{note}
+In [TMAP8 V&V](https://mooseframework.inl.gov/TMAP8/verification_and_validation/ver-1e.html) there is a note stating that the expressions of the analytical solution for the transient case in TMAP4 and TMAP7 V&V books are inconsistent with the results, which suggest typographical errors. Therefore, we used the expression provided in TMAP8 V&V and taken from {cite}`li2010analytical`. However, we assume that there is also a typogaphical error in the TMAP8 case description as a reasonable agreement of numerical and analytical results can be obtained by replacing the minus sign before the summations with the plus sign.
+```
+
+```{code-cell} ipython3
+import festim as F
+import numpy as np
+
+from dolfinx.geometry import bb_tree, compute_colliding_cells, compute_collisions_points
+
+
+class PointValue(F.VolumeQuantity):
+    def __init__(self, field, volume, x0, filename=None):
+        super().__init__(field, volume, filename)
+        self.x0 = x0
+
+    def compute(self):
+        u = self.field.solution
+        mesh = u.function_space.mesh
+        tree = bb_tree(mesh, mesh.geometry.dim)
+        cell_candidates = compute_collisions_points(tree, self.x0)
+        cell = compute_colliding_cells(mesh, cell_candidates, self.x0).array
+        assert len(cell) > 0
+        first_cell = cell[0]
+
+        self.value = self.field.solution.eval(self.x0, first_cell)
+        self.data.append(self.value)
+
+
+class Profile(F.VolumeQuantity):
+    def __init__(self, field, volume, times=None, filename=None):
+        super().__init__(field, volume, filename)
+        self.times = times or []
+
+    def compute(self):
+        self.value = self.field.solution.x.array[:].copy()
+        self.data.append(self.value)
+```
+
+## FESTIM Code
+
+```{code-cell} ipython3
+a = 33e-6
+l = 66e-6
+D0_PyC = 1.274e-7
+D0_SiC = 2.622e-11
+T = 1.e3
+c_left = 3.0537e25
+
+model = F.HydrogenTransportProblem()
```

Comment:
Note: using this class in a multimaterial case works because you have continuity of concentration... Making this case not super relevant as far as physics are concerned

Reply:
Nah it’s fine